api/bucket.go

```diff
@@ -1,19 +1,209 @@
 package api
 
 import (
-	"github.com/gin-gonic/gin"
-	"github.com/konveyor/tackle2-hub/model"
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
 	"io"
 	"net/http"
 	"os"
+	"path"
 	pathlib "path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/tackle2-hub/model"
 )
 
 //
 // BucketHandler provides bucket management.
 type BucketHandler struct {
 }
 
+func (h *BucketHandler) serveBucketGet(ctx *gin.Context, owner *model.BucketOwner) {
+	if ctx.Request.Header.Get(Directory) == DirectoryArchive {
+		h.getDirArchive(ctx, path.Join(owner.Bucket, ctx.Param(Wildcard)))
```

Comment:
Was thinking about the Wildcard validation to not allow "applications/1/bucket/../.." GET or PUT, but this directory traversal is handled by gin router which "resolves" ../.. as part of URL parsing which results to "applications/".